scripts/quests/flyers_for_regine.lua

```diff
@@ -0,0 +1,89 @@
+
+quests = quests or {}
+quests.flyers_for_regine = quests.flyers_for_regine or {}
+
+-------------------------------------------------
+-- local data
+-------------------------------------------------
+
+-- [ffr]deliveryMask stores delivered fliers as bits (0 = not delivered, 1 = delivered)
+-- the key in this table relates to the bit in the mask
+-- region is the region in the zone around the NPC, for trigging the approach message
+-- offset is the message that plays when approaching the NPC (offset from FFR_LOOKS_CURIOUSLY_BASE)
+
+local npcData =
+{
+    [ 0] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 1, x =   76.331, z = -128.655, offset = 0, npc = 'Answald'},
+    [ 1] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 2, x =  -13.768, z =  -95.857, offset = 1, npc = 'Prietta'},
+    [ 2] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 3, x =    0.750, z =  -81.438, offset = 2, npc = 'Miene'},
+    [ 3] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 4, x =  -21.999, z = -106.877, offset = 3, npc = 'Portaure'},
+    [ 4] = {zone = tpz.zone.PORT_SAN_DORIA,     region = 5, x =  -37.821, z =   40.949, offset = 4, npc = 'Auvare'},
+
+    [ 5] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 2, x =  146.420, z =  127.601, offset = 0, npc = 'Coullene'},
+    [ 6] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 3, x = -159.082, z =  253.794, offset = 1, npc = 'Guilberdrier'},
+    [ 7] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 4, x = -156.242, z =   49.184, offset = 2, npc = 'Boncort'},
+    [ 8] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 5, x = -127.355, z =  130.461, offset = 3, npc = 'Capiria'},
+    [ 9] = {zone = tpz.zone.NORTHERN_SAN_DORIA, region = 6, x = -157.524, z =  263.818, offset = 4, npc = 'Villion'},
+
+    [10] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 2, x =   33.033, z =  -30.119, offset = 0, npc = 'Blendare'},
+    [11] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 3, x =   69.895, z =   41.073, offset = 1, npc = 'Rosel'},
+    [12] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 4, x =  105.147, z =  -16.735, offset = 3, npc = 'Maugie'},
+    [13] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 5, x =   80.378, z =  -24.644, offset = 5, npc = 'Adaunel'},
+    [14] = {zone = tpz.zone.SOUTHERN_SAN_DORIA, region = 6, x = -136.322, z =   21.958, offset = 7, npc = 'Leuveret'},
+}
+
+-------------------------------------------------
+-- public functions
+-------------------------------------------------
+
+quests.flyers_for_regine.initZone = function(zone)
+    local zoneId = zone:getID()
+
+    for k, v in pairs(npcData) do
+        if v.zone == zoneId then
+            zone:registerRegion(v.region, v.x, 10, v.z, 0, 0, 0)
+        end
+    end
+end
+
+quests.flyers_for_regine.onRegionEnter = function(player, region)
+    local zoneId = player:getZoneID()
+    local regionId = region:GetRegionID()
+
+    for k, v in pairs(npcData) do
+        if v.zone == zoneId and v.region == regionId then
```

Comment:
regionEnter checks could then use the offsets too, so you're only checking up to 5 regions per check, as opposed to potentially up to 15.